### PR TITLE
Add ktor 1.1.5

### PR DESCRIPTION
**Pull request by norac89**
Before submitting your PR, please review the following checklist :

## If you are adding a framework

+ [x] Please check configuration files
   + [x] `neph.yml`, for [neph](http://tbrand.github.io/neph) job processing
   + [x] `FRAMEWORKS.yml`, for **frameworks** list
   + [x] `travis.yml`, for CI
+ [x] A `Dockerfile` exists ?
+ [x] All tests passes ?
~~~
export FRAMEWORK=<MY_FRAMEWORK>
shards install
shards build --release
bin/neph ${FRAMEWORK}
crystal spec
~~~

Adds [Ktor](https://ktor.io/)

### CODE REVIEW

**Comment by waghanza** [comment]
Awesome :tada: the first framework in `kotlin`

A pleasure to **review**

File: FRAMEWORKS.yml [hunk 1]
```diff
@@ -323,3 +323,8 @@ c:
     website: github.com/ohler55/agoo-c
     version: "0.4"
     language: "11"
+kotlin:
+  ktor:
+    website: ktor.io
+    version: "1.1.5"
```
Comment (waghanza) [inline FRAMEWORKS.yml]: Could you use `X.Y` format here ? No needs to be **string**

File: FRAMEWORKS.yml [hunk 2]
```diff
@@ -323,3 +323,8 @@ c:
...
+    language: "1.3.31"
```
Comment (waghanza) [inline FRAMEWORKS.yml]: Could you use `X.Y` format here ? No needs to be **string**

File: kotlin/ktor/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+FROM gradle:4.10-jdk8
```
Comment (waghanza) [inline kotlin/ktor/Dockerfile]: Why this version ? The latest available is not the `5` ?

File: kotlin/ktor/resources/logback.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%d{YYYY-MM-dd HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+    <root level="trace">
```
Comment (waghanza) [inline kotlin/ktor/resources/logback.xml]: Could it be disabled ?
All frameworks has lot disabled here (or at least **SHOULD**)

File: kotlin/ktor/Dockerfile (hunk 3 above)
Comment (norac89) [reply kotlin/ktor/Dockerfile]: Had problem in a more complex previous project updating to gradle 5. So just out of lazyness.
Made the change here and everything went smoothly.

**Comment by waghanza** [comment]
@norac89 I can see that https://hub.docker.com/_/gradle use `jdk` **8** and https://hub.docker.com/r/zenika/kotlin `jdk` **12**

Any reason to choose the first ? 

It seems that `kotlin` **COULD** target `jdk` **12**

* https://github.com/keeganwitt/docker-gradle/issues/91
* https://github.com/docker-library/openjdk/issues/302

**Comment by norac89** [comment]
From https://blog.jetbrains.com/kotlin/2019/04/kotlin-1-3-30-released/

> If you run the code under JVM versions 9, 10, 11, or 12, you can now set the corresponding jvmTarget. This setting will affect the version of the generated classfiles, and the resulting bytecode won’t run on any lower version of JVM. Note that so far newer versions don’t add any bytecode optimizations or features beyond the ones that exist in lower versions, but that is going to change in the future.

I think that the code is compile to JVM 8 bytecode by default. Haven't tested targeting other JVM version yet. But it seems there would be no gain doing so for now.

I must say I am no Kotlin expert. Came from python and started playing with it recently. So I may be wrong...

Anyway I have updated the docker image to jdk11 (the latest version from the official gradle images)

File: kotlin/ktor/resources/application.conf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+ktor {
+    deployment {
+        port = 3000
+        port = ${?PORT}
```
Comment (waghanza) [inline kotlin/ktor/resources/application.conf]: Not a duplicate ?

File: FRAMEWORKS.yml [hunk 6]
```diff
@@ -323,3 +323,8 @@ c:
     website: github.com/ohler55/agoo-c
     version: "0.4"
     language: "11"
+kotlin:
+  ktor:
+    website: ktor.io
+    version: 1.1.5
```
Comment (waghanza) [inline FRAMEWORKS.yml]: Could you use only `X.Y` ? I mean `1.1`, this is more readable for audience

File: FRAMEWORKS.yml [hunk 7]
```diff
@@ -323,3 +323,8 @@ c:
...
+    language: 1.3.31
```
Comment (waghanza) [inline FRAMEWORKS.yml]: Same as above

File: kotlin/ktor/src/Application.kt [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+fun main(args: Array<String>): Unit = io.ktor.server.netty.EngineMain.main(args)
+
+@Suppress("unused") // Referenced in application.conf
+fun Application.module() {
+    routing {
+        get("/") {
+            call.respondText("", contentType = ContentType.Text.Plain, status = HttpStatusCode.OK)
+        }
+        get("/user/{id}") {
```
Comment (waghanza) [inline kotlin/ktor/src/Application.kt]: Could this route be constrains ?
I mean can we for `id` to be numeric ?

**Comment by waghanza** [comment]
No skills in `kotlin` either -> will have to test ;-)

**Comment by waghanza** [comment]
Here is a table of perf recorded on my 4 CPU x 8 Gb (what seems to say that performance decrease for `jdk` **11**).

Let's wait for https://github.com/keeganwitt/docker-gradle/pull/98 to be release ;-) to add `jdk` **12** here

|  | Average | 50th percentile | 90th percentile | 99th percentile | 99.9th percentile | Standard deviation | Req / s | Throughput |
|----------------------------------------|----------------:|----------------:|----------------:|----------------:|----------------:|----------------:|-----------:|----------------:| 
| jdk8 | 26.48 ms | 21.60 ms | 46.48 ms | 89.58 ms | 331.00 ms | 18494.00 | 38843.67 | 60.59 MB |
| jdk11 | 38.63 ms | 28.73 ms | 70.78 ms | 183.35 ms | 709.34 ms | 39578.67 | 28423.00 | 44.30 MB |

**Comment by norac89** [comment]
Interesting results. Did not think it would be that slower on jdk 11. 
But kotlin code is still compiled to JVM 8 bytecode even if the jdk 11 is used. There is a `jvmTarget` setting that allows kotlin to be compile to specific JVM version. But haven't play with it yet. And I don't think I will have the time, at least in the next weeks.

I think it would be better to keep everything running on a jdk 8 image which correspond to the generated bytecode version. Also, documentation on ktor is using jdk 8.

File: kotlin/ktor/resources/application.conf (hunk 5 above)
Comment (norac89) [reply kotlin/ktor/resources/application.conf]: It allows to override the port if a environment variable PORT is present.

File: kotlin/ktor/src/Application.kt (hunk 8 above)
Comment (norac89) [reply kotlin/ktor/src/Application.kt]: 3817050

File: FRAMEWORKS.yml (hunk 7 above)
Comment (norac89) [reply FRAMEWORKS.yml]: 9faf6c1

File: FRAMEWORKS.yml (hunk 6 above)
Comment (norac89) [reply FRAMEWORKS.yml]: 9faf6c1

File: kotlin/ktor/resources/application.conf (hunk 5 above)
Comment (waghanza) [reply kotlin/ktor/resources/application.conf]: ok

**Comment by waghanza** [comment]
@norac89 I've updated results with `jdk` **11**, will consider to use `jdk` **8**

**Comment by waghanza** [comment]
@fracaron do you have any recommendation for `kotlin` lint ? I'm try to use community standard for all frameworks here in https://github.com/the-benchmarker/web-frameworks/pull/1011

**Comment by fracaron** [comment]
I haven't used any lint tools yet except for what is included in Intellij IDE. 
With a quick search, I found https://ktlint.github.io/.

**Comment by waghanza** [comment]
Yeah that's the one I've used in https://github.com/waghanza/http-benchmark/blob/community_standard/.ci/lint.sh#L170

**Pull request merged**